### ISSUE TITLE
Union types inconsistency

### ISSUE THREAD

**Issue by ml054** [Microsoft/TypeScript#9982]
**TypeScript Version:**  1.8.0 

**Code**

```ts

type Sex = 'male' | 'female';
	
interface Person {
	Sex: Sex;
	Age: number;
}

// this DOESN'T work
var p1:Person = {
	Sex: "male"
} as Person;

// it works:
var p2: Person = {
	Age: 8
} as Person;

// it works (but is ugly)
var p3: Person = {
	Sex: <Sex> "male"
} as Person;
```

**Expected behavior:**
Should compile
**Actual behavior:** 
Doesn't compile

**Comment by yortus** [comment]
The sample code compiles fine for me in TS 2.0 beta.

**Comment by RyanCavanaugh** [comment]
Confirmed fixed in 2.0
